cmd/pulumi.go

```diff
@@ -76,6 +82,24 @@ func NewPulumiCmd() *cobra.Command {
 				}
 			}
 
+			if rsp, err := http.Get("https://pulumi.io/latest-version"); err == nil && rsp.StatusCode == http.StatusOK {
+				buf := bytes.Buffer{}
+				buf.ReadFrom(rsp.Body)
+
+				latestVer, latestErr := semver.ParseTolerant(buf.String())
+				curVer, curErr := semver.ParseTolerant(version.Version)
+
+				if latestErr == nil && curErr == nil {
+					if latestErr == nil && curErr == nil && !isDevVersion(curVer) {
+						if latestVer.GT(curVer) {
+							cmdutil.Diag().Warningf(
+								diag.Message("", "Your current version of Pulumi is out of date. The latest "+
+									"version is '%s', while you're on '%s'."), latestVer, curVer)
```

Comment:
Should we include an actionable link - perhaps to `https://pulumi.io/install`?

Reply:
Hmm, why not just print the `curl` command for mac/Linux and PowerShell command for Windows?

Reply:
For now that may make sense - once we support additional installation approaches (or in case folks like to download zip manually) - it wouldn't hurt to indirect through an install page.

This is sort of responding to the point you raised about not plastering `curl | bash` everywhere and instead funneling folks though a high quality install page (even if that page itself still leans on a curl install command).

Reply:
+1 to Luke's comments.

Reply:
This is different though. You already installed the product. There's no need to indirect in that case, presumably you're using it and loving it and just want to update. For that, we should print the specific gesture needed to accomplish that.

I think NPM, Yarn, and others are exemplars here.

Reply:
Note that we do already have two ways to install - curl|bash and manual zip.  Not clear that folks who opt for the later on initial install will be okay with he former on upgrade. In the future we will likely have more options. 

I guess we could do something crazy like Yarn:

https://github.com/yarnpkg/yarn/blob/cf3a5f214648c84a839fe09065795904f2688997/src/cli/commands/install.js#L76-L106

Reply:
If we do want to make a change here, I’m happy to use the following huristic: If the location on disk of the Pulumi binary is ~/.pulumi/bin/pulumi, print the curl command. Similar for Windows. Otherwise just print a link to the pulumi.io page.

I do worry that this is going to end up causing us to have a super long warning (especially in the case of Windows, where the automated install is a mouthful)

Reply:
Yarn's approach here may be crazy, but it leads to a great user experience. That's the thing to optimize for here. The entire point of doing this work is to get people to upgrade more often in as friction-free a manner as possible. As a user, I would be enormously frustrated by a nag message that forced me to launch a web browser and read some freeform text to figure out how to make it happy. We absolutely **must** print the actionable, copy-and-pastable message.